File: todo_backend/todo_backend/Models/Todo.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using todo_backend.Enums;

namespace todo_backend.Models
{
    public class Todo
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; } = String.Empty;

        [BsonElement("title")]
        public string Title { get; set; } = String.Empty;

        [BsonElement("description")]
        public string? Description { get; set; }

        [BsonElement("status")]
        public TodoStatus Status { get; set; } = TodoStatus.Pending;
    }
}

File: frontend/src/models/Todo.ts
import { TodoStatus } from "../enums/TodoStatus";

export interface Todo {
  id: string;
  title: string;
  description: string;
  status: TodoStatus;
}

export interface CreateTodo {
  title: string;
  description?: string;
  status: TodoStatus;
}

export interface UpdateTodo {
  title: string;
  description?: string;
  status: TodoStatus;
}